zowex/src/main.rs

```diff
@@ -279,30 +290,48 @@ fn run_daemon_command(args: &mut Vec<String>) -> io::Result<()> {
         _resp.append(&mut stdin);
     }
 
-    // form our host, port, and connection strings
-    let daemon_host = "127.0.0.1".to_owned();
-    let port_string = get_port_string();
+    let mut tries = 0;
+    loop {
+        let socket_string = get_socket_string();
+        let mut stream = establish_connection(socket_string)?;
+        match talk(&_resp, &mut stream) {
+            Ok(result) => { return Ok(result); },
+            Err(ref e) if e.kind() == io::ErrorKind::ConnectionReset => {
+                if tries > THREE_MIN_OF_RETRIES {
+                    println!("Terminating after {} connection retries.", THREE_MIN_OF_RETRIES);
+                    std::process::exit(EXIT_CODE_TIMEOUT_CONNECT_TO_RUNNING_DAEMON);
+                }
 
-    let mut stream = establish_connection(daemon_host, port_string)?;
-    talk(&_resp, &mut stream)
+                println!("{} ({} of {})", "The Zowe daemon is in use, retrying", tries, THREE_MIN_OF_RETRIES);
+                
+                // pause between attempts to connect
+                thread::sleep(Duration::from_secs(THREE_SEC_DELAY));
+                tries += 1;
+            },
+            Err(e) => { return Err(e); }
```

Comment:
Instead of matching the `Ok` and `Err` halves of the Result type separately, do we want to combine them into one?
```
match talk(&_resp, &mut stream) {
    Err(ref e) if e.kind() == io::ErrorKind::ConnectionReset => {
        ...
    },
    result => { return result; },
}
```

Reply:
Done